src/Akka.Persistence.Sql/Journal/Dao/BaseByteArrayJournalDao.cs

```diff
@@ -112,22 +112,37 @@ public async Task<IImmutableList<Exception>> AsyncWriteMessages(
 
         public async Task Delete(string persistenceId, long maxSequenceNr)
         {
+            long maxMarkedDeletion;
+
+            // no need to use transaction
+            await using (var connection = ConnectionFactory.GetConnection())
+            {
+                maxMarkedDeletion = await MaxMarkedForDeletionMaxPersistenceIdQuery(connection, persistenceId, maxSequenceNr).FirstOrDefaultAsync(ShutdownToken);
+            }
+
+            if (maxMarkedDeletion is 0)
+                return;
+            
             await ConnectionFactory.ExecuteWithTransactionAsync(
                 WriteIsolationLevel,
                 ShutdownToken,
                 async (connection, token) =>
                 {
-                    await connection
-                        .GetTable<JournalRow>()
+                    var journalTable = connection.GetTable<JournalRow>();
+                    await journalTable
                         .Where(
                             r =>
                                 r.PersistenceId == persistenceId &&
-                                r.SequenceNumber <= maxSequenceNr)
+                                r.SequenceNumber == maxMarkedDeletion)
```

Comment:
Instead of marking all of the records less than the requested sequence number as deleted, we mark the highest affected instead. Update 1 record instead of many.